sympy/sets/fancysets.py

```diff
@@ -483,6 +483,128 @@ def _intersect(self, other):
+
+        >>> from sympy import Lambda
+        >>> from sympy.sets.fancysets import ImageSet
+        >>> from sympy import pprint, Union
+        >>> from sympy.core import Dummy, pi, S, Symbol
+        >>> n = Dummy('n')
+        >>> img1 = ImageSet(Lambda(n, 2*n*pi + 3*pi/4), S.Integers)
+        >>> img2 = ImageSet(Lambda(n, 2*n*pi + 7*pi/4), S.Integers)
+        >>> pprint(img1._union(img2), use_unicode=False)
+                3*pi
+        {n*pi + ---- | n in Integers()}
+                 4
+
+        """
+
+        from sympy.core import pi
+        from sympy.polys import Poly
+        from sympy.simplify import simplify
+        from sympy.polys.polyerrors import PolynomialError
+
+        if other is S.EmptySet:
+            return self
+
+        def first_val(imgset):
+            base = imgset.base_set
+            if base == S.Reals:
+                return imgset.lamda(0)
+            try:
+                iterable = iter(base)
+                return imgset.lamda(next(iterable))
+            except TypeError:
+                raise TypeError
+
+        def _attempt_1(self_expr, final_expr, var_self, var_final, base):
+            var_term = self_expr.coeff(var_self)
+            if var_term == 0:
+                raise NotImplementedError
+            if simplify(
+                    final_expr.subs(var_final, var_self) -
+                    self_expr) % var_term == var_term / 2:
+                # (a*pi + pi + expr1) - ( a*pi  + exp1) == abs(a/2)*pi
+                # can be => (a/2)*pi + expr1
+                final_expr = (var_term / 2) * var_self + first_val(self)
+                ans = ImageSet(Lambda(var_self, final_expr), base)
+            elif simplify(
+                    final_expr -
+                    self_expr) % var_term == 0:
+                # (a*pi + expr1) - ( a*pi  + 2*pi + exp1)= 0
+                # can be => a*pi + expr1
+                final_expr = var_term * var_self + first_val(self)
+                ans = ImageSet(Lambda(var_self, final_expr), base)
+            else:
+                ans = None
+            return ans
+
+        if isinstance(other, ImageSet):
+            try:
+                base = other.base_set
+                base_self = self.base_set
+                if base != base_self:
+                    raise NotImplementedError
+                # use one dummy variable n_self
+                var_self = self.lamda.variables
+                if len(var_self) > 1:
+                    # Not Implemented for multiple lambda variables.
+                    return None
+                var_self = var_self[0]
+                var_final = other.lamda.variables
+                if len(var_final) > 1:
+                    # Not Implemented for multiple lambda variables.
+                    return None
+                var_final = var_final[0]
+
+                # Extracting expr
+                lamb_expr = self.lamda.expr
+                if Poly(lamb_expr, var_self).is_linear:
```

Comment:
Can you put the `try-except` block just around this statement ?

Its clearer that way